plugins/pulp_rpm/plugins/migrations/0045_populate_vendor.py

```diff
@@ -0,0 +1,66 @@
+import gzip
+import sys
+
+from pulp.server.db.connection import get_collection
+from pulp.server.db.migrations.lib import utils
+
+if sys.version_info < (2, 7):
+    from xml.etree import ElementTree as ET
+else:
+    from xml.etree import cElementTree as ET
+
+
+_NAMESPACES = {
+    'common': "http://linux.duke.edu/metadata/common",
+    'rpm': 'http://linux.duke.edu/metadata/rpm',
+}
+
+# The unit metadata snippets miss header/encoding and namespace references
+# which aren't included until publish time. This breaks the parsing.
+_HEADER = ('<?mxl version="1.0" encoding="UTF-8"?>\n'
+           '<metadata xmlns="http://linux.duke.edu/metadata/common" '
+           'xmlns:rpm="http://linux.duke.edu/metadata/rpm">\n {}'
+           '</metadata>\n')
+
+
+def migrate_rpm(collection, unit):
+    """
+    Uncompress single rpm unit primary metadata and populate the unit vendor attribute
+    accordingly.
+
+    :param collection: a collection of RPM units
+    :type collection: pymongo.collection.Collection
+    :param unit: the RPM unit being migrated
+    :type unit: dict
+    """
+    primary_xml = unit.get('repodata', {}).get('primary', '')
+    primary_xml = _HEADER.format(gzip.zlib.decompress(primary_xml))
+    root_element = ET.fromstring(primary_xml)
+    delta = {}
+    delta['vendor'] = root_element.find(
```

Comment:
The only lines that change in this migrations vs. [0044_populate_recommends.py](https://github.com/pulp/pulp_rpm/blob/2-master/plugins/pulp_rpm/plugins/migrations/0044_populate_recommends.py#L42) are these lines and line 60